webapp/lib/firebase_tools.dart

```diff
@@ -39,8 +39,42 @@ Dataset loadDataset(String datasetName) {
   if (VERBOSE) print("Loading dataset: $datasetName");
 
   // Temporary code
-  if (datasetName != 'test-dataset') {
+  if (datasetName == null) {
     throw new DatasetLoadException('Sorry, dataset "$datasetName" not available to load.');
   }
-  return new Dataset.fromJson(jsonDatasetTwoSchemesNoCodes);
+
+  String msgCountDatasetPrefix = 'dataset-msg-';
+  if (datasetName.startsWith(msgCountDatasetPrefix)) {
+    try {
+      int count = int.parse(datasetName.replaceFirst(msgCountDatasetPrefix, ''));
+      return generateEmptyDataset(datasetName, 3, count);
+    } catch (e) {
+      throw new DatasetLoadException('Sorry, dataset "$datasetName" not available to load.');
+    }
+  }
+  if (datasetName == 'test-dataset') {
+    return new Dataset.fromJson(jsonDatasetTwoSchemesNoCodes);
+  }
+  throw new DatasetLoadException('Sorry, dataset "$datasetName" not available to load.');
+}
+
+Dataset generateEmptyDataset(String name, int schemeCount, int messageCount) {
```

Comment:
I don't think this really belongs in firebase tools - please could you move it to a new data-set-tools file